### PR TITLE
[DEN-165] Expand buildrequest method to handle query and path params

### PR DESCRIPTION
**Pull request by ssaluja4**

### CODE REVIEW

File: edgecast/client/client.go [hunk 1]
```diff
@@ -132,6 +133,82 @@ func (c Client) BuildRequest(
...
+		}
+	}
+
+	req, err := retryablehttp.NewRequest(method, absoluteURL.String(), payload)
+
+	if err != nil {
+		return nil, fmt.Errorf("BuildRequest: NewRequest: %v", err)
+	}
+
+	if body != nil {
```
Comment (changgyu-oh) [inline edgecast/client/client.go:193]: If body is a string type, should it be "text/html"?

**Review by changgyu-oh — COMMENTED** [review]
I left a comment.

File: edgecast/client/client.go (hunk 1 above)
Comment (ssaluja4) [reply edgecast/client/client.go:195]: Updated to add content type as "text/plain; charset=utf-8", when bodytype is string

Comment (ssaluja4) [reply edgecast/client/client.go:195]: Updated to add content type "text/plain; charset=utf-8" when body is of type string

**Review by changgyu-oh — APPROVED** [review]
Looks good!

**Pull request merged**